INCode/diagramconfiguration.py

```diff
@@ -107,28 +122,57 @@ def reveal_children(self):
             child_tree_item = CallableTreeItem(child, current_item)
             child_tree_item.setExpanded(True)
 
+    def init_preview(self):
+        if self.svg_view_.isVisible():
+            content = self.generate_uml()
+            if content:
+                self.load_view_signal.emit(content)
+
     def export(self):
-        content = self.entry_point_item_.export()
-        self.generate_uml(content)
+        print('exporting ', self.entry_point_item_.callable.name)
+        content = self.generate_uml()
+        self.load_svg_view(content)
 
-    def toggle_uml(self):
-        if self.image_wrapper_.isVisible():
-            self.image_wrapper_.hide()
+    def toggle_preview(self):
+        if self.svg_view_.isVisible():
+            self.svg_view_.hide()
         else:
-            self.image_wrapper_.show()
-
-    def generate_uml(self, content):
-        timestamp = datetime.datetime.now().strftime("%Y%m%d%H%M%S")
-        temp_file_name = os.path.join(self.temp_dir_, timestamp)
-        file = open(temp_file_name, "w+")
-        file.write(content)
-        file.close()
-        subprocess.call(["plantuml", temp_file_name])
-        subprocess.call(["rm", temp_file_name])
-
-        temp_file_name += ".png"
-        pixmap = QPixmap(temp_file_name)
-        self.image_.setPixmap(pixmap)
-        self.image_.resize(pixmap.size())
-        subprocess.call(["rm", temp_file_name])
-        self.image_wrapper_.show()
+            self.svg_view_.show()
+
+    def toggle_layout(self):
+        orientation = Qt.Vertical if self.wrapper.orientation() == Qt.Horizontal else Qt.Horizontal
+        self.wrapper.setOrientation(orientation)
+
+    def generate_uml(self, content=None):
+        if not content:
+            content = self.entry_point_item_.export()
+        if content == self.current_diagram_ or content == '@startuml\n\n\n@enduml':
+            return
```

Comment:
[Python guidelines](https://www.python.org/dev/peps/pep-0008/#id51) discourage inconsistent return types: "Be consistent in return statements. Either all return statements in a function should return an expression, or none of them should."